src/circuit/gadget/ecc/chip.rs

```diff
@@ -0,0 +1,361 @@
+use super::EccInstructions;
+use crate::circuit::gadget::utilities::{copy, CellValue, Var};
+use crate::constants::{self, OrchardFixedBasesFull, ValueCommitV, NUM_WINDOWS, NUM_WINDOWS_SHORT};
+use arrayvec::ArrayVec;
+use ff::{Field, PrimeFieldBits};
+use halo2::{
+    arithmetic::CurveAffine,
+    circuit::{Chip, Layouter},
+    plonk::{Advice, Column, ConstraintSystem, Error, Fixed, Permutation, Selector},
+};
+use std::marker::PhantomData;
+
+pub(super) mod add;
+pub(super) mod add_incomplete;
+pub(super) mod mul;
+pub(super) mod mul_fixed;
+pub(super) mod witness_point;
+pub(super) mod witness_scalar_fixed;
+
+/// A curve point represented in affine (x, y) coordinates. Each coordinate is
+/// assigned to a cell.
+#[derive(Clone, Debug)]
+pub struct EccPoint<C: CurveAffine> {
+    /// x-coordinate
+    pub x: CellValue<C::Base>,
+    /// y-coordinate
+    pub y: CellValue<C::Base>,
+}
+
+impl<C: CurveAffine> EccPoint<C> {
+    /// Returns the value of this curve point, if known.
+    pub fn point(&self) -> Option<C> {
+        match (self.x.value(), self.y.value()) {
+            (Some(x), Some(y)) => {
+                if x == C::Base::zero() && y == C::Base::zero() {
+                    Some(C::identity())
+                } else {
+                    Some(C::from_xy(x, y).unwrap())
+                }
+            }
+            _ => None,
+        }
+    }
```

Comment:
```suggestion
    }

    pub fn x(&self) -> CellValue<C::Base> {
        self.x
    }

    pub fn y(&self) -> CellValue<C::Base> {
        self.y
    }
```

Reply:
Should these getters be gadget-level? We currently have `extract_p()` to get the x-coordinate at the gadget-level.

Within the chip, each submodule can access x and y anyway.